pgcli/pgexecute.py

```diff
@@ -471,6 +481,14 @@ def databases(self):
             cur.execute(self.databases_query)
             return [x[0] for x in cur.fetchall()]
 
+    def full_databases(self):
+        with self.conn.cursor() as cur:
+            _logger.debug('Databases Query. sql: %r',
+                          self.full_databases_query)
+            cur.execute(self.full_databases_query)
+            headers = [x[0] for x in cur.description]
+            return cur.fetchall(), headers, cur.statusmessage
```

Comment:
Instead of fetching all the rows, you can just return the cursor and `format_output()` will consume the cursor as it goes through each row.